core/src/main/java/org/infinispan/marshall/core/internal/BytesEncoding.java

```diff
@@ -0,0 +1,406 @@
+package org.infinispan.marshall.core.internal;
+
+/**
+ * Basic encoding/decoding of primitives
+ */
+final class BytesEncoding implements Encoding<BytesObjectOutput, BytesObjectInput> {
```

Comment:
+1 to remove the encoding interface+class and do the encoding directly in the ByteObjectOutput/Input classes.